controlplane/kubeadm/internal/cluster_test.go

```diff
@@ -178,6 +154,178 @@ func TestGetMachinesForCluster(t *testing.T) {
 	g.Expect(machines).To(HaveLen(1))
 }
 
+func TestGetWorkloadCluster(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	// create an envtest for all subtests
+	testEnv := helpers.NewTestEnvironment()
+	defer func() {
+		if testEnv != nil {
+			g.Expect(testEnv.Stop()).To(Succeed(), "stopping envtest")
+		}
+	}()
```

Comment:
It would be nice if `helpers.NewTestEnvironment` would return a cleanup function along with `testEnv`, so we don't have to do this in every test.